e2e_testing/torchscript/elementwise.py

```diff
@@ -419,6 +419,194 @@ def ElementwiseGtIntTensorModule_basic(module, tu: TestUtils):
+    ])
+    def forward(self, x, y):
+        return torch.lt(x, y)
+
+
+@register_test_case(module_factory=lambda: ElementwiseLtFloatTensorModule())
+def ElementwiseLtFloatTensorModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 5), tu.rand(5))
+
+
+class ElementwiseLtIntTensorModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.int64, True),
+        ([-1], torch.int64, True),
+    ])
+    def forward(self, x, y):
+        return torch.lt(x, y)
+
+
+@register_test_case(module_factory=lambda: ElementwiseLtIntTensorModule())
+def ElementwiseLtIntTensorModule_basic(module, tu: TestUtils):
+    module.forward(torch.randint(10, (3, 5)), torch.randint(10, (5,)))
+
+# ==============================================================================
+
+class ElementwiseEqFloatScalarModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+    ])
+    def forward(self, x):
+        return torch.eq(x, 6.0)
+
+
+@register_test_case(module_factory=lambda: ElementwiseEqFloatScalarModule())
+def ElementwiseEqFloatScalarModule_basic(module, tu: TestUtils):
+    module.forward(torch.tensor([[1.0, 2.2, 6.0], [6.0, 2.0, 3.1]])
+                   .to(torch.float32))
+
+
+class ElementwiseEqIntScalarModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.int64, True),
+    ])
+    def forward(self, x):
+        return torch.eq(x, 2)
+
+
+@register_test_case(module_factory=lambda: ElementwiseEqIntScalarModule())
+def ElementwiseEqIntScalarModule_basic(module, tu: TestUtils):
+    module.forward(torch.randint(2, 4, (5,8)))
+
+
+class ElementwiseEqDiffWidthScalarModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.int32, True),
+    ])
+    def forward(self, x):
+        return torch.eq(x, 2)
+
+
+@register_test_case(module_factory=lambda: ElementwiseEqDiffWidthScalarModule())
+def ElementwiseEqDiffWidthScalarModule_basic(module, tu: TestUtils):
+    module.forward(torch.randint(2, 4, (5,8)).to(torch.int32))
+
+
+class ElementwiseEqFloatTensorModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+        ([-1], torch.float32, True),
+    ])
+    def forward(self, x, y):
+        return torch.eq(x, y)
+
+
+@register_test_case(module_factory=lambda: ElementwiseEqFloatTensorModule())
+def ElementwiseEqFloatTensorModule_basic(module, tu: TestUtils):
+    module.forward(torch.tensor([[1.0, 2.2, 6.0], [6.0, 2.0, 3.1]])
+                   .to(torch.float32), 
+                   torch.tensor([1.0, 2.4, 6.0]).to(torch.float32))
+
```

Comment:
Why aren't you passing random tensors?

Reply:
There is a high possibility that two random floating-point tensors won't collide, thus producing always false result. So it's better to have unit test cases having same values at some indices.